kv/kvserver: TestReplicateQueueRebalanceMultiStore failed

kv/kvserver.TestReplicateQueueRebalanceMultiStore [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12246819?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12246819?buildTab=artifacts#/) on master @ [d2c5bbe1aa7930d8e7267357f051ba7d2a79aec2](https://github.com/cockroachdb/cockroach/commits/d2c5bbe1aa7930d8e7267357f051ba7d2a79aec2):

Fatal error:

```
panic: test timed out after 14m55s
running tests:
	TestReplicateQueueRebalanceMultiStore (14m33s)
	TestReplicateQueueRebalanceMultiStore/multi-store (14m23s)

```
Stack: 

```
goroutine 16028085 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2241 +0x31c
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
* sync.runtime_SemacquireMutex(0x4012cf30c0?, 0x0?, 0x4013bcf398?)
* 	GOROOT/src/runtime/sema.go:77 +0x28
* sync.(*Mutex).lockSlow(0x400f8ded58)
* 	GOROOT/src/sync/mutex.go:171 +0x174
* sync.(*Mutex).Lock(...)
* 	GOROOT/src/sync/mutex.go:90
* github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Replica).RangeFeed(0x400f8dec80, 0x4018e68160, {0xffff7a1df8d8, 0x40042cb300}, 0x2?)
* 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/replica_rangefeed.go:266 +0x2fc
* github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).RangeFeed(0x40066af500, 0x4018e68160, {0xffff7a1df8d8, 0x40042cb300})
* 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/store.go:3000 +0xe0
* github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Stores).RangeFeed(0x75bed00?, 0x4018e68160, {0xffff7a1df8d8, 0x40042cb300})
* 	github.com/cockroachdb/cockroach/pkg/kv/kvserver/pkg/kv/kvserver/stores.go:227 +0xf0
* github.com/cockroachdb/cockroach/pkg/server.(*Node).RangeFeed(0x400bf47400, 0x4018e68160, {0x7603f40?, 0x40042cb300})
* 	github.com/cockroachdb/cockroach/pkg/server/node.go:1673 +0x1fc
* github.com/cockroachdb/cockroach/pkg/rpc.internalClientAdapter.RangeFeed.func2.1({0x0?, 0x75bed00?}, {0x75f7140?, 0x4012cf3080?})
* 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:1032 +0x78
* github.com/cockroachdb/cockroach/pkg/util/tracing/grpcinterceptor.StreamServerInterceptor.func1({0x5cc8240, 0x400bf47400}, {0x75f3558?, 0x4000fb9560?}, 0xab7aa60, 0x40075e16c0)
* 	github.com/cockroachdb/cockroach/pkg/util/tracing/grpcinterceptor/grpc_interceptor.go:165 +0x4e8
* github.com/cockroachdb/cockroach/pkg/rpc.serverStreamInterceptorsChain.run.func1({0x5cc8240?, 0x400bf47400?}, {0x75f3558?, 0x4000fb9560?})
* 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:847 +0x60
* github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func4({0x5cc8240, 0x400bf47400}, {0x75f3558, 0x4000fb9560}, 0x400c22c000?, 0x4010aedb40)
* 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:178 +0x78
* github.com/cockroachdb/cockroach/pkg/rpc.serverStreamInterceptorsChain.run.func1({0x5cc8240?, 0x400bf47400?}, {0x75f3558?, 0x4000fb9560?})
* 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:850 +0x8c
* github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.streamInterceptor({0x400c22c000?, {{0x75bec58?}, {0x75fdf50?, 0x4016831110?}}}, {0x5cc8240, 0x400bf47400}, {0x75f3558, 0x4000fb9560}, 0xab7aa60, 0x4010aedb40)
* 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/auth.go:167 +0x30c
* github.com/cockroachdb/cockroach/pkg/rpc.serverStreamInterceptorsChain.run.func1({0x5cc8240?, 0x400bf47400?}, {0x75f3558?, 0x4000fb9560?})
* 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:850 +0x8c
* github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func2.1({0x4008ecadb8?, 0x463330?})
* 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:145 +0x34
* github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr(0x400c073320, {0x75bed00, 0x4000fb9530}, {0x75bec58?, 0x4008ecaea8?}, 0x4008ecae68)
* 	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:336 +0xac
* github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func2({0x5cc8240, 0x400bf47400}, {0x75f3558?, 0x4000fb9560?}, 0xab7aa60, 0x4010aedb40)
* 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:144 +0xcc
* github.com/cockroachdb/cockroach/pkg/rpc.serverStreamInterceptorsChain.run.func1({0x5cc8240?, 0x400bf47400?}, {0x75f3558?, 0x4000fb9560?})
* 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:850 +0x8c
* github.com/cockroachdb/cockroach/pkg/rpc.serverStreamInterceptorsChain.run({0x40050989a0, 0x4, 0x4}, {0x5cc8240, 0x400bf47400}, {0x75f3558, 0x4000fb9560}, 0xab7aa60, 0x40075e16c0)
* 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:852 +0x12c
* github.com/cockroachdb/cockroach/pkg/rpc.internalClientAdapter.RangeFeed.func2()
* 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:1037 +0xfc
* created by github.com/cockroachdb/cockroach/pkg/rpc.internalClientAdapter.RangeFeed
* 	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:1025 +0x2a0
* 
* goroutine 14276796 [select]:
* github.com/cockroachdb/pebble/vfs.(*diskHealthCheckingFile).startTicker.func1()
* 	github.com/cockroachdb/pebble/vfs/external/com_github_cockroachdb_pebble/vfs/disk_health.go:171 +0x9c
* created by github.com/cockroachdb/pebble/vfs.(*diskHealthCheckingFile).startTicker
* 	github.com/cockroachdb/pebble/vfs/external/com_github_cockroachdb_pebble/vfs/disk_health.go:166 +0x64
* 
*
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReplicateQueueRebalanceMultiStore.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-32533

kv/kvserver.TestReplicateQueueRebalanceMultiStore [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12247136?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12247136?buildTab=artifacts#/) on master @ [d2c5bbe1aa7930d8e7267357f051ba7d2a79aec2](https://github.com/cockroachdb/cockroach/commits/d2c5bbe1aa7930d8e7267357f051ba7d2a79aec2):

Fatal error:

```
panic: test timed out after 14m55s
running tests:
	TestReplicateQueueRebalanceMultiStore (14m33s)
	TestReplicateQueueRebalanceMultiStore/multi-store (14m11s)

```
Stack: 

```
goroutine 92894645 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2241 +0x31c
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestReplicateQueueRebalanceMultiStore
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/7334302ae8c22802de8334e77d6901c9/logTestReplicateQueueRebalanceMultiStore3579821348
    test_log_scope.go:81: use -show-logs to present logs inline
=== RUN   TestReplicateQueueRebalanceMultiStore/multi-store
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReplicateQueueRebalanceMultiStore.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver.TestReplicateQueueRebalanceMultiStore [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12254519?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12254519?buildTab=artifacts#/) on master @ [eb148c0d58a5acd941c26d4f317624a9d9601217](https://github.com/cockroachdb/cockroach/commits/eb148c0d58a5acd941c26d4f317624a9d9601217):

Fatal error:

```
panic: test timed out after 14m55s
running tests:
	TestReplicateQueueRebalanceMultiStore (14m27s)
	TestReplicateQueueRebalanceMultiStore/multi-store (14m14s)

```
Stack: 

```
goroutine 94025724 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2241 +0x31c
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestReplicateQueueRebalanceMultiStore
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/7334302ae8c22802de8334e77d6901c9/logTestReplicateQueueRebalanceMultiStore2949399704
    test_log_scope.go:81: use -show-logs to present logs inline
=== RUN   TestReplicateQueueRebalanceMultiStore/multi-store
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReplicateQueueRebalanceMultiStore.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvserver.TestReplicateQueueRebalanceMultiStore [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12255810?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12255810?buildTab=artifacts#/) on master @ [eb148c0d58a5acd941c26d4f317624a9d9601217](https://github.com/cockroachdb/cockroach/commits/eb148c0d58a5acd941c26d4f317624a9d9601217):

Fatal error:

```
panic: test timed out after 14m55s
running tests:
	TestReplicateQueueRebalanceMultiStore (14m37s)
	TestReplicateQueueRebalanceMultiStore/multi-store (14m27s)

```
Stack: 

```
goroutine 53062544 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2241 +0x31c
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
    replicate_queue_test.go:339: current lease state (want at least 10 leases on all stores): [19 14 11 25 8]
    replicate_queue_test.go:328: current replica state (want at least 59 replicas on all stores): [67 67 63 67 64]
    replicate_queue_test.go:339: current lease state (want at least 10 leases on all stores): [19 14 11 25 8]
    replicate_queue_test.go:328: current replica state (want at least 59 replicas on all stores): [67 67 63 67 64]
    replicate_queue_test.go:339: current lease state (want at least 10 leases on all stores): [19 14 11 25 8]
    replicate_queue_test.go:328: current replica state (want at least 59 replicas on all stores): [67 67 63 67 64]
    replicate_queue_test.go:339: current lease state (want at least 10 leases on all stores): [19 14 11 25 8]
    replicate_queue_test.go:328: current replica state (want at least 59 replicas on all stores): [67 67 63 67 64]
    replicate_queue_test.go:339: current lease state (want at least 10 leases on all stores): [19 14 11 25 8]
    replicate_queue_test.go:328: current replica state (want at least 59 replicas on all stores): [67 67 63 67 64]
    replicate_queue_test.go:339: current lease state (want at least 10 leases on all stores): [19 14 11 25 8]
    replicate_queue_test.go:328: current replica state (want at least 59 replicas on all stores): [67 67 63 67 64]
    replicate_queue_test.go:339: current lease state (want at least 10 leases on all stores): [19 14 11 25 8]
    replicate_queue_test.go:328: current replica state (want at least 59 replicas on all stores): [67 67 63 67 64]
    replicate_queue_test.go:339: current lease state (want at least 10 leases on all stores): [19 14 11 25 8]
    replicate_queue_test.go:328: current replica state (want at least 59 replicas on all stores): [67 67 63 67 64]
    replicate_queue_test.go:339: current lease state (want at least 10 leases on all stores): [19 14 11 25 8]
    replicate_queue_test.go:328: current replica state (want at least 59 replicas on all stores): [67 67 63 67 64]
    replicate_queue_test.go:339: current lease state (want at least 10 leases on all stores): [19 14 11 25 8]
    replicate_queue_test.go:328: current replica state (want at least 59 replicas on all stores): [67 67 63 67 64]
    replicate_queue_test.go:339: current lease state (want at least 10 leases on all stores): [19 14 11 25 8]
    replicate_queue_test.go:328: current replica state (want at least 59 replicas on all stores): [67 67 63 67 64]
    replicate_queue_test.go:339: current lease state (want at least 10 leases on all stores): [19 14 11 25 8]
    replicate_queue_test.go:328: current replica state (want at least 59 replicas on all stores): [67 67 63 67 64]
    replicate_queue_test.go:339: current lease state (want at least 10 leases on all stores): [19 14 11 25 8]
    replicate_queue_test.go:328: current replica state (want at least 59 replicas on all stores): [67 67 63 67 64]
    replicate_queue_test.go:339: current lease state (want at least 10 leases on all stores): [19 14 11 25 8]
    replicate_queue_test.go:328: current replica state (want at least 59 replicas on all stores): [67 67 63 67 64]
    replicate_queue_test.go:339: current lease state (want at least 10 leases on all stores): [19 14 11 25 8]
    replicate_queue_test.go:328: current replica state (want at least 59 replicas on all stores): [67 67 63 67 64]
    replicate_queue_test.go:339: current lease state (want at least 10 leases on all stores): [19 14 11 25 8]
    replicate_queue_test.go:328: current replica state (want at least 58 replicas on all stores): [67 67 63 66 64]
    replicate_queue_test.go:339: current lease state (want at least 10 leases on all stores): [19 14 11 25 8]
    replicate_queue_test.go:328: current replica state (want at least 58 replicas on all stores): [67 67 63 66 64]
    replicate_queue_test.go:339: current lease state (want at least 10 leases on all stores): [19 14 11 25 8]
    replicate_queue_test.go:328: current replica state (want at least 58 replicas on all stores): [67 67 63 66 64]
    replicate_queue_test.go:339: current lease state (want at least 10 leases on all stores): [19 14 11 25 8]
    replicate_queue_test.go:328: current replica state (want at least 58 replicas on all stores): [67 67 63 66 64]
    replicate_queue_test.go:339: current lease state (want at least 10 leases on all stores): [19 14 11 25 8]
    replicate_queue_test.go:328: current replica state (want at least 58 replicas on all stores): [67 67 63 66 64]
    replicate_queue_test.go:339: current lease state (want at least 10 leases on all stores): [19 14 11 25 8]
    replicate_queue_test.go:328: current replica state (want at least 58 replicas on all stores): [67 67 63 66 64]
    replicate_queue_test.go:339: current lease state (want at least 10 leases on all stores): [19 14 11 25 8]
    replicate_queue_test.go:328: current replica state (want at least 65 replicas on all stores): [68 76 70 76 73]
    replicate_queue_test.go:339: current lease state (want at least 10 leases on all stores): [18 15 11 25 8]
    replicate_queue_test.go:328: current replica state (want at least 65 replicas on all stores): [71 75 70 74 73]
    replicate_queue_test.go:339: current lease state (want at least 10 leases on all stores): [20 15 11 22 9]
    replicate_queue_test.go:328: current replica state (want at least 65 replicas on all stores): [71 75 70 74 73]
    replicate_queue_test.go:339: current lease state (want at least 10 leases on all stores): [19 14 11 21 12]
=== RUN   TestReplicateQueueRebalanceMultiStore/multi-store
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReplicateQueueRebalanceMultiStore.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/server/init.go
+++ b/pkg/server/init.go
@@ -731,7 +731,7 @@ func inspectEngines(
 		initializedEngines = append(initializedEngines, eng)
 	}
 	clusterVersion, err := kvstorage.SynthesizeClusterVersionFromEngines(
-		ctx, engines, binaryVersion, binaryMinSupportedVersion,
+		ctx, initializedEngines, binaryVersion, binaryMinSupportedVersion,
 	)
 	if err != nil {
 		return nil, err

